### PR TITLE
trivial: allow compiling without systemd system call filter support

### PR DESCRIPTION
**Pull request by superm1**
Closes: https://github.com/fwupd/fwupd/issues/8078

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by superm1** [comment]
CC @computerix

File: meson_options.txt [hunk 1]
```diff
@@ -328,6 +328,11 @@ option('systemd_unit_user',
   value: 'fwupd-refresh',
   description: 'User account to use for fwupd-refresh.service (empty for DynamicUser)',
 )
+option('systemd_syscall_filter',
+  type: 'boolean',
+  value: 'true',
+  description: 'Whether to enable systemd syscall filter',
```
Comment (hughsie) [inline meson_options.txt:334]: I think just "Enable systemd...

**Comment by computerix** [comment]
After explicitly setting systemd_syscall_filter to false in gentoo's ebuild and applying the patch above, 
fwupd builds and works for me with a systemd without syscall filtering
(as expected, the build installs a fwupd.service without the syscall filtering lines).
journalctl also looks fine.

File: meson_options.txt (hunk 1 above)
Comment (superm1) [reply meson_options.txt]: ack, fixed.

**Pull request merged**